src/borg/archiver/extract_cmd.py

```diff
@@ -56,32 +56,42 @@ def do_extract(self, args, repository, manifest, archive):
         else:
             pi = None
 
-        for item in archive.iter_items(filter, preload=True):
+        all_items = list(archive.iter_items(preload=True))
+
+        for item in all_items:
             orig_path = item.path
             if strip_components:
                 item.path = os.sep.join(orig_path.split(os.sep)[strip_components:])
+
             if not args.dry_run:
                 while dirs and not item.path.startswith(dirs[-1].path):
                     dir_item = dirs.pop(-1)
                     try:
                         archive.extract_item(dir_item, stdout=stdout)
                     except BackupError as e:
                         self.print_warning_instance(BackupWarning(remove_surrogates(dir_item.path), e))
-            if output_list:
-                logging.getLogger("borg.output.list").info(remove_surrogates(item.path))
+
+                if output_list and matcher.match(item.path):
+                    logging.getLogger("borg.output.list").info(remove_surrogates(item.path))
```

Comment:
this log output is different from the other one, no status char.

do you practically try your code after making changes?

Reply:
you also can run the unit tests locally, just invoke pytest or tox.

Reply:
@ThomasWaldmann ?